File: 2nd Year/1st Semester/MAP/Laborator/SEM1_LAB2_MAP/factory/TaskContainerFactory.java
package sem1_2.factory;

import sem1_2.model.Container;
import sem1_2.model.QueueContainer;
import sem1_2.model.StackContainer;

public class TaskContainerFactory implements Factory {

    private static final TaskContainerFactory instance = new TaskContainerFactory();

    public static TaskContainerFactory getInstance() {
        return instance;
    }

    @Override
    public Container createContainer(Strategy strategy) {
        return switch (strategy) {
            case FIFO -> new QueueContainer();
            case LIFO -> new StackContainer();
        };
    }
}

File: 2nd Year/1st Semester/MAP/Laborator/Lab9/Lab9/factory/TaskContainerFactory.cs
using System.Diagnostics;
using Lab9.model;

namespace Lab9.factory;

public class TaskContainerFactory : IFactory
{
    
    private static readonly TaskContainerFactory _instance = new TaskContainerFactory();

    public static TaskContainerFactory GetInstance()
    {
        return _instance;
    }
    
    public IContainer CreateContainer(Strategy strategy)
    {
        if (strategy == Strategy.Fifo)
            return new QueueContainer();
        return new StackContainer();
    }
}